### PR TITLE
Update map() and optimize() documentation

### PR DESCRIPTION
**Pull request by senarvi**
Fixes documentation related to #260 .

Adds notes that the types and list sizes must not changed between samples, when calling `optimize()`.

When looking at the code, I noticed that the documentation of the inputs seems to be outdated. AFAICS, inputs can be any sequence or a streaming dataloder, so I updated the type annotation and removed this from the docstring: "Each input should contain at least a valid filepath". Let me know if I missed something, or feel free to make changes in the PR.

My code editor also removed unnecessary whitespace. I guess that's good.

### CODE REVIEW

**Comment by deependujha** [comment]
Hi @senarvi, thanks for opting to contribute to litdata.

Btw, Can you please convert your PR into draft until it's ready for review?

**Comment by senarvi** [comment]
@deependujha how do I convert a PR into a draft?

**Comment by senarvi** [comment]
Does anyone have an idea why the Windows and MacOS tests fail with "Numpy is not available"?

**Comment by deependujha** [comment]
> @deependujha how do I convert a PR into a draft?

Screenshot from 2024-07-24 14-41-31

Just click on convert to Draft.

---

I'll try attempting the failing tests soon.

**Comment by senarvi** [comment]
Thanks @deependujha , I haven't noticed that link! That will be useful in the future. From my behalf this PR is ready. I think anything in this PR is not responsible for breaking the tests, since this doesn't include functional changes. So better to leave this open, or what do you think?

**Comment by deependujha** [comment]
sure. I'm looking into it. Thanks for contributing.

**Review by tchaton — APPROVED** [review]

File: requirements/test.txt [hunk 1]
```diff
@@ -10,3 +10,4 @@ pandas
 lightning
 lightning-cloud == 0.5.70 # Must be pinned to ensure compatibility
 zstd
+numpy < 2.0
```
Comment (deependujha) [inline requirements/test.txt:13]: Why this though?
I believe a new PR was merged recently which added support for numpy>=2!

Comment (deependujha) [inline requirements/test.txt:13]: Why this though?
I believe a new PR was merged recently which added support for numpy>=2!

**Review by deependujha — COMMENTED** [review]
All failing tests were bcoz of numpy. But, recently support for numpy>=2 was added.

File: requirements/test.txt (hunk 1 above)
Comment (tchaton) [reply requirements/test.txt:13]: Yes.

Comment (deependujha) [reply requirements/test.txt:13]: nice answer. 😀
for the first part or second part (hack vs correct implementation)

**Pull request merged**